js/AudioWorkletGlobalScope.js

```diff
@@ -21,14 +23,72 @@ const kWorkletInputs = Symbol.for('node-web-audio-api:worklet-inputs');
 const kWorkletOutputs = Symbol.for('node-web-audio-api:worklet-outputs');
 const kWorkletParams = Symbol.for('node-web-audio-api:worklet-params');
 const kWorkletParamsCache = Symbol.for('node-web-audio-api:worklet-params-cache');
+const kWorkletGetBuffer = Symbol.for('node-web-audio-api:worklet-get-buffer');
+const kWorkletRecycleBuffer = Symbol.for('node-web-audio-api:worklet-recycle-buffer');
+const kWorkletRecycleBuffer1 = Symbol.for('node-web-audio-api:worklet-recycle-buffer-1');
+const kWorkletMarkAsUntransferable = Symbol.for('node-web-audio-api:worklet-mark-as-untransferable');
 // const kWorkletOrderedParamNames = Symbol.for('node-web-audio-api:worklet-ordered-param-names');
 
+
 const nameProcessorCtorMap = new Map();
 const processors = {};
 let pendingProcessorConstructionData = null;
 let loopStarted = false;
 let runLoopImmediateId = null;
 
+class BufferPool {
+  #bufferSize;
+  #pool;
+
+  constructor(bufferSize, initialPoolSize) {
+    this.#bufferSize = bufferSize;
+    this.#pool = new Array(initialPoolSize);
+
+    for (let i = 0; i < this.#pool.length; i++) {
+      this.#pool[i] = this.#allocate();
+    }
+  }
+
+  #allocate() {
+    const float32 = new Float32Array(this.#bufferSize);
+    markAsUntransferable(float32);
+    // Mark underlying buffer as untransfrable too, this will fail one of
+    // the task in `audioworkletprocessor-process-frozen-array.https.html`
+    // but prevent segmentation fault
+    markAsUntransferable(float32.buffer);
+
+    return float32;
+  }
+
+  get() {
+    if (this.#pool.length === 0) {
+      return this.#allocate();
+    }
+
+    return this.#pool.pop();
+  }
+
+  recycle(buffer) {
```

Comment:
We don't need to set the values to zero because on each render call we copy all input values anyway?

Reply:
Yup!